modules/components/pages/outputs/azure_data_lake_gen2.adoc

```diff
@@ -0,0 +1,123 @@
+
+ifndef::env-cloud[]
+Introduced in version 4.38.0.
+endif::[]
+
+
+```yml
+# Configuration fields, showing default values
+output:
+  label: ""
+  azure_data_lake_gen2:
+    storage_account: "" # No default (optional)
+    storage_access_key: "" # No default (optional)
+    storage_connection_string: "" # No default (optional)
+    storage_sas_token: "" # No default (optional)
+    filesystem: messages-${!timestamp("2006")} # No default (required)
+    path: ${!counter()}-${!timestamp_unix_nano()}.txt
+    max_in_flight: 64
+```
+To specify a different <<path,`path` value>> (file name) for each file, use xref:configuration:interpolation.adoc#bloblang-queries[function
+interpolations]. Function interpolations are calculated for each message in a batch.
+
+== Authentication methods
+
+This output supports multiple authentication methods. You must configure at least one method from the following list:
+
+- `storage_connection_string`
+- `storage_account` and `storage_access_key`
+- `storage_account` and `storage_sas_token`
+- `storage_account` to access using https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#DefaultAzureCredential[DefaultAzureCredential^]
+
+If you configure multiple authentication methods, the `storage_connection_string` takes precedence.
+
+== Performance
+
+This output benefits from sending multiple messages in flight in parallel for improved performance. You can tune the max number of in flight messages (or message batches) with the field `max_in_flight`.
+
+== Fields
+
+=== `storage_account`
+
+The storage account to access. This field is ignored when the `storage_connection_string` field is populated.
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `storage_access_key`
+
+The access key for the storage account. Use this field along with `storage_account` for authentication. This field is ignored when the `storage_connection_string` field is populated.
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `storage_connection_string`
+
+The connection string for the storage account. You must enter a value for this field if no other authentication method is specified.
+
+NOTE: If the `storage_connection_string` field does not contain the `AccountName` parameter value, specify it in the
+`storage_account` field.
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `storage_sas_token`
+
+The SAS token for the storage account. Use this field along with `storage_account` for authentication. This field is ignored when either the `storage_connection_string` or `storage_access_key` fields are populated.
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `filesystem`
+
+The name of the data lake storage file system you want to upload messages to. This field supports xref:configuration:interpolation.adoc#bloblang-queries[interpolation functions].
+
+*Type*: `string`
+
+```yml
+# Examples
+filesystem: messages-${!timestamp("2006")}
```

Comment:
Wouldn't you still need to include the name of the filesystem? I had a quick flick through the Microsoft docs but couldn't easily find where to get this/it's format.

Reply:
The example would generate filesystem names like `messages-2024` which is a valid filesystem name.

The format of the field can be found in [the REST API documentation](https://learn.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/filesystem/create?view=rest-storageservices-datalakestoragegen2-2019-12-12) for the API call to create a filesystem.

> The filesystem identifier. The value must start and end with a letter or number and must contain only letters, numbers, and the dash (-) character. Consecutive dashes are not permitted. All letters must be lowercase. The value must have between 3 and 63 characters.

Similar to the `storage_account` and related fields, filesystem names have the same requirements as the [`container`](https://docs.redpanda.com/redpanda-connect/components/inputs/azure_blob_storage/#container) field in the `aws_blob_storage` plugin.

For a little context, the Azure Data Lake Storage Gen2 service is built upon the Blob Storage service, so there are direct analogues between the two services:
- `storage_account` -> `storage_account`
- `container` -> `filesystem`
- `path` -> `path`